gqcp/tests/QCMethod/CC/QCMethod_CCD_test.cpp

```diff
@@ -0,0 +1,108 @@
+    const auto rhf_qc_structure = GQCP::QCMethod::RHF<double>().optimize(objective, plain_rhf_scf_solver, rhf_environment);
+    const auto rhf_parameters = rhf_qc_structure.groundStateParameters();
+
+    r_spinor_basis.transform(rhf_parameters.coefficientMatrix());
+
+    // Check if the intermediate RHF results are correct. We can't continue if this isn't the case.
+    const auto rhf_energy = rhf_qc_structure.groundStateEnergy() + GQCP::Operator::NuclearRepulsion(molecule).value();
+    const double ref_rhf_energy = -74.942079928192;
+    BOOST_REQUIRE(std::abs(rhf_energy - ref_rhf_energy) < 1.0e-09);
+
+
+    // Create a GSpinorBasis since we have implement spinor-CCD, and quantize the molecular Hamiltonian in it.
+    const auto g_spinor_basis = GQCP::GSpinorBasis<double, GQCP::GTOShell>::FromRestricted(r_spinor_basis);
+    const auto M = g_spinor_basis.numberOfSpinors();
+    const auto g_sq_hamiltonian = GQCP::SQHamiltonian<double>::Molecular(g_spinor_basis, molecule);  // in the canonical restricted spin-orbitals
+
+    // Create the GHF ONV (which is actually just the RHF ONV, since we're using the canonical RHF orbitals) and the corresponding orbital space.
+    //const auto reference_onv = GQCP::SpinUnresolvedONV::GHF(2 * K, N, rhf_parameters.spinOrbitalEnergiesBlocked());
+    //const auto orbital_space = reference_onv.orbitalSpace();
+    // Use a manually-made orbital space (#FIXME in develop).
+    const auto orbital_space = GQCP::OrbitalSpace({0, 1, 2, 3, 4, 7, 8, 9, 10, 11}, {5, 6, 12, 13});  // occupied and virtual indices
+
+
+    BOOST_REQUIRE(orbital_space.numberOfOrbitals() == M);
+
+
+    // Initialize an environment suitable for CCD.
+    auto environment_ccd = GQCP::CCSDEnvironment<double>::PerturbativeCCD(g_sq_hamiltonian, orbital_space);
+    auto environment_ccsd_ref = GQCP::CCSDEnvironment<double>::PerturbativeCCSD(g_sq_hamiltonian, orbital_space);
+
+    // Functional step that sets the T1-amplitudes to zero, needed for our reference CCD solver.
+    //GQCP::FunctionalStep<GQCP::CCSDEnvironment<double>> set_T1_zero([GQCP::OrbitalSpace orbital_space](GQCP::CCSDEnvironment<double> environment_ccsd_ref){
+    //   environment_ccsd_ref.t1_amplitudes.back() = GQCP::T1Amplitudes<double>(orbital_space.initializeRepresentableObjectFor<double>(GQCP::OccupationType::k_occupied, GQCP::OccupationType::k_virtual), orbital_space);
+    //    }, "Set the T1-amplitudes to zero.");
+    GQCP::FunctionalStep<GQCP::CCSDEnvironment<double>> set_T1_zero([](GQCP::CCSDEnvironment<double>& environment_ccsd_ref){
+        const auto& orbital_space = environment_ccsd_ref.t1_amplitudes.back().orbitalSpace();
+        environment_ccsd_ref.t1_amplitudes.back() = GQCP::T1Amplitudes<double>(environment_ccsd_ref.t1_amplitudes.back().orbitalSpace().initializeRepresentableObjectFor<double>(GQCP::OccupationType::k_occupied, GQCP::OccupationType::k_virtual), orbital_space);
+    }, "Set the T1-amplitudes to zero.");
```

Comment:
Great job on this one!

Since it's not a special kind of constructor, I think it's better to use the `{}` constructor, as in the following suggestion:
```suggestion
    GQCP::FunctionalStep<GQCP::CCSDEnvironment<double>> set_T1_zero {[](GQCP::CCSDEnvironment<double>& environment_ccsd_ref){
        const auto& orbital_space = environment_ccsd_ref.t1_amplitudes.back().orbitalSpace();
        environment_ccsd_ref.t1_amplitudes.back() = GQCP::T1Amplitudes<double>(environment_ccsd_ref.t1_amplitudes.back().orbitalSpace().initializeRepresentableObjectFor<double>(GQCP::OccupationType::k_occupied, GQCP::OccupationType::k_virtual), orbital_space);
    }, "Set the T1-amplitudes to zero."};
```